### PR TITLE
Strategy names and PEP8

### PR DESCRIPTION
**Pull request by meatballs**
- Ensures that each Strategy class has a unique name (required for the API)
- Fixes some PEP8 errors

### CODE REVIEW

File: axelrod/strategies/cycler.py [hunk 1]
```diff
@@ -67,7 +68,7 @@ def __init__(self, cycle="CCD"):
         """
         Player.__init__(self)
         self.cycle = cycle
-        self.name += " " + cycle
+        self.name = "Cycler {0}".format(cycle)
```
Comment (marcharper) [inline axelrod/strategies/cycler.py]: FYI you can leave off the 0 in `{0}`.

File: axelrod/strategies/cycler.py [hunk 2]
```diff
@@ -77,6 +78,8 @@ def strategy(self, opponent):
 
 
 class CyclerDC(Cycler):
+
+    name = 'Cycler DC'
```
Comment (marcharper) [inline axelrod/strategies/cycler.py]: Do we need to set the name explicitly in addition to the dynamic assignment above?

File: axelrod/strategies/gobymajority.py [hunk 3]
```diff
@@ -80,6 +82,7 @@ class GoByMajority40(GoByMajority):
     """
     GoByMajority player with a memory of 40.
     """
+    name = 'Go By Majority 40'
```
Comment (marcharper) [inline axelrod/strategies/gobymajority.py]: Same question -- is the explicit name needed?

File: axelrod/tests/unit/test_player.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import copy
+# -*- coding: utf-8 -*-import copy
```
Comment (marcharper) [inline axelrod/tests/unit/test_player.py]: I think this isn't quite right...

File: axelrod/tests/unit/test_player.py [hunk 5]
```diff
@@ -259,7 +259,7 @@ def classifier_test(self, expected_class_classifier=None):
         self.assertTrue('memory_depth' in player.classifier,
                         msg="memory_depth not in classifier")
         self.assertTrue('stochastic' in player.classifier,
-                        msg="stochastic not in classifier")
+                        msg="stochastic n§ot in classifier")
```
Comment (marcharper) [inline axelrod/tests/unit/test_player.py]: Did you want that extra character?

File: axelrod/strategies/cycler.py (hunk 1 above)
Comment (meatballs) [inline axelrod/strategies/cycler.py]: Yep - will do

File: axelrod/strategies/cycler.py (hunk 2 above)
Comment (meatballs) [inline axelrod/strategies/cycler.py]: Yes. the dynamic assignment only works at instance level and the API needs a unique id for each class.

File: axelrod/strategies/gobymajority.py (hunk 3 above)
Comment (meatballs) [inline axelrod/strategies/gobymajority.py]: same response!

File: axelrod/tests/unit/test_player.py (hunk 4 above)
Comment (meatballs) [inline axelrod/tests/unit/test_player.py]: you're right. It isn't. I'll sort it.

File: axelrod/tests/unit/test_player.py (hunk 5 above)
Comment (meatballs) [inline axelrod/tests/unit/test_player.py]: nope!

**Comment by drvinceknight** [comment]
> Ensures that each Strategy class has a unique name (required for the API)

Should/could we have a test for this? (Sorry)

**Comment by meatballs** [comment]
> Should/could we have a test for this? (Sorry)

yes, probably - I had these tests on #694 which is where theses issues were highlighted in the first place.

**Comment by meatballs** [comment]
I think I've covered everything on this one. Anything I've missed?

**Comment by drvinceknight** [comment]
👍

**Pull request merged**